examples/arithmetics/src/language-server/arithmetics-scope-provider.ts

```diff
@@ -17,7 +18,7 @@ export class ArithmeticsScopeProvider extends DefaultScopeProvider {
     }
 
     protected override getGlobalScope(referenceType: string): Scope {
-        return new StreamScope(this.indexManager.allElements(referenceType), undefined, { caseInsensitive: true });
+        return new MapScope(this.indexManager.allElements(referenceType), undefined, { caseInsensitive: true });
```

Comment:
A MapScope does not bring any performance improvement when used without caching because our standard usage of scopes traverses the scope only once.